app/ui/src/app/integrations/edit-page/action-select/action-select.component.ts

```diff
@@ -102,22 +103,35 @@ export class IntegrationsSelectActionComponent extends FlowPage
   }
 
   ngOnInit() {
-    this.actions = this.connector
-      .filter(connector => connector !== undefined)
-      .map(connector => connector.actions);
+    // set a local var for current step
+    this.currentStep = +this.route.snapshot.paramMap.get('position');
+
+    // if it's a start step
+    if (this.currentStep === this.currentFlow.getFirstPosition()) {
+      this.actions = this.connector
+        .filter(connector => connector !== undefined)
+        .switchMap(connector => [connector.actions.filter(action => action.pattern === 'From')]);
+    }
+
+    // if it's anything other than a start step
+    if (this.currentStep > this.currentFlow.getFirstPosition()
+      && this.currentStep <= this.currentFlow.getLastPosition()) {
+      this.actions = this.connector
+        .filter(connector => connector !== undefined)
+        .switchMap(connector => [connector.actions.filter(action => action.pattern === 'To')]);
```

Comment:
Here we're replicating code in two blocks just to check 2 two different types of value for the same variable. This is not DRY. Please take a look into RxJS *lettable operators* as a way to remove code duplication.

Also, you can replace `.filter(connector => connector !== undefined)` by the broader `.filter(connector => !!connector)`